Make `test_distributed_spawn.py` tell you how to run it correctly

wconstab opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #112924

Sample output if incorrect/missing args are specified:

```
RuntimeError: Missing expected env vars for `test_distributed_spawn.py`.  Please
ensure to specify the following:
'BACKEND' = one of ('gloo', 'nccl', 'ucc')
'WORLD_SIZE' = int >= 2
'TEMP_DIR' specifying a directory containing a barrier file named
'barrier'.

e.g.
touch /tmp/barrier && TEMP_DIR=/tmp BACKEND='nccl' WORLD_SIZE=2 python
/data/users/whc/pytorch/test/distributed/test_distributed_spawn.py
```

wconstab committed:
Make `test_distributed_spawn.py` tell you how to run it correctly

Sample output if incorrect/missing args are specified:

'''
Missing expected env vars for `test_distributed_spawn.py`.  Please
ensure to specify the following:
'BACKEND' = one of ('gloo', 'nccl', 'ucc')
'WORLD_SIZE' = int >= 2
'TEMP_DIR' specifying a directory containing a barrier file named
'barrier'.

e.g.
touch /tmp/barrier && TEMP_DIR=/tmp BACKEND='nccl' WORLD_SIZE=2 python
/data/users/whc/pytorch/test/distributed/test_distributed_spawn.py
'''

[ghstack-poisoned]

voznesenskym commented on test/distributed/test_distributed_spawn.py:
```diff
@@ -25,9 +26,26 @@
+        "'TEMP_DIR' specifying a directory containing a barrier file named 'barrier'.\n\n"
+        f"e.g.\ntouch /tmp/barrier && TEMP_DIR=/tmp BACKEND='nccl' WORLD_SIZE=2 python {__file__}",
+    )
+    sys.exit(0)
```
instead of print and exit(0) (0 means good), raise?

wconstab committed:
Update on "Make `test_distributed_spawn.py` tell you how to run it correctly"


Sample output if incorrect/missing args are specified:

```
RuntimeError: Missing expected env vars for `test_distributed_spawn.py`.  Please
ensure to specify the following:
'BACKEND' = one of ('gloo', 'nccl', 'ucc')
'WORLD_SIZE' = int >= 2
'TEMP_DIR' specifying a directory containing a barrier file named
'barrier'.

e.g.
touch /tmp/barrier && TEMP_DIR=/tmp BACKEND='nccl' WORLD_SIZE=2 python
/data/users/whc/pytorch/test/distributed/test_distributed_spawn.py
```

[ghstack-poisoned]

wanchaol commented on test/distributed/test_distributed_spawn.py:
```diff
@@ -25,9 +26,25 @@
+        "Missing expected env vars for `test_distributed_spawn.py`.  Please ensure to specify the following:\n"
+        f"'BACKEND' = one of {_allowed_backends}\n"
+        f"'WORLD_SIZE' = int >= 2\n"
+        "'TEMP_DIR' specifying a directory containing a barrier file named 'barrier'.\n\n"
```
maybe make the error message clear that `barrier` is inter-test barrier not the dist.barrier?

wconstab replied:
oops i didn't notice this, well, i'm gonna leave it alone but if you want to change it ill stamp ;)

wanchaol reviewed (approved):
Nice!

wconstab commented:
@pytorchbot merge

wz337 commented:
Thx for adding this. I need this desperately! So I don't have to go look how to run this test everytime.

fduwjj reviewed (commented):
Well, how to run the test is documented here: https://github.com/pytorch/pytorch/blob/main/torch/distributed/CONTRIBUTING.md.

Maybe we should just put the link in the beginning of test? (And I 100% agree it was confusing and frustrating experience to figure out how to run this test...)
